### PR TITLE
images/cilium-test: New test suite image

### PR DESCRIPTION
**Pull request by errordeveloper**
This is to be used with GKE test cluster operator that will be
introduce in the due course. The operator will be in charge of
running the tests, hence there needs to be an image with all
test binaries and assets included.

### CODE REVIEW

File: images/cilium-test/go.mod
Comment (aanm) [inline images/cilium-test/go.mod]: perhaps go 1.15?

**Review by aanm — COMMENTED** [review]
Just a small comment. Adding @tklauser to the reviewers list since he might have some ideas to pin the go version for all images.

File: images/cilium-test/cilium-test-gke.sh
Comment (christarazi) [inline images/cilium-test/cilium-test-gke.sh:28]: Nit: These lines can be consolidated:

```suggestion
export CILIUM_IMAGE="${1}"
export CILIUM_OPERATOR_IMAGE="${2}"
```

There are a few other occurrences of this.

**Review by christarazi — APPROVED** [review]
LGTM, just a minor nit. Doesn't require a second review.

**Review by tklauser — APPROVED** [review]
Regarding the Go version: other than the `go.mod` file I don't see it specified anywhere in this PR. For these I think we should agree on when to bump it (i.e. do we bump it whenever a new 1.x version is released or only if there are language changes). Since we use `1.15` in the root `go.mod` I think it should be changed in `images/cilium-test/go.mod` as well as pointed out by @aanm.

Otherwise LGTM.

File: images/cilium-test/cilium-test-gke.sh
Comment (errordeveloper) [reply images/cilium-test/cilium-test-gke.sh:28]: The reason I ended up doing this is because shellcheck compained:
```
In ./cilium-test/cilium-test-gke.sh line 32:
export K8S_VERSION="$(kubectl version -o json |  jq -r '(.serverVersion.major + "." + (.serverVersion.minor | scan("[0-9]+")))' | sed 's/"//g')"
       ^---------^ SC2155: Declare and assign separately to avoid masking return values.
```
Looks like it's not necessary for each of the variables to have separate `export` line, only the one with subshell, but I just find that way it's more consistent overall.

**Pull request merged**
